src/cmd/add-module/main.go

```diff
@@ -45,7 +45,7 @@ func main() {
 
 	err = func() error {
 		// Lower case input
-		re := regexp.MustCompile("(?P<Namespace>[a-zA-Z0-9]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)")
+		re := regexp.MustCompile("(?P<Namespace>[a-zA-Z0-9-]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)")
```

Comment:
@cam72cam shouldn't this include a `/` prefix in the regexp itself and at least a `$` at the end? This regexp may match incomplete or incorrect repositories.

Reply:
Thanks for the fast feedback!

In case we use `^` and `$` to match the whole line, we also need to iterate over the comment line by line.

e.g.

```go
re := regexp.MustCompile("^(?P<Namespace>[a-zA-Z0-9-]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)$")
repoScanner := bufio.NewScanner(strings.NewReader(*repository))
var match []string
for len(match) == 0 && repoScanner.Scan() {
   match = re.FindStringSubmatch(repoScanner.Text())
}
```


Alternatives for matching the namespace taken from this repository:
* `[a-zA-Z0-9-]+` current _(too specific?)_
* `.*` in `internal/v1api/providers.go` `func ArchivedOverrides` _(too broad?)_
* `[^/]+` in `internal/provider/list.go` `providerDirectoryRegex` _(my favorite)_


Happy to implement your decision.

Reply:
Thank you for all your help! I'm unfortunately not as well versed in the registry code as @cam72cam so I'll let him make the call on this. He should be online in a bit hopefully.

Reply:
This should work fine.  The match below requires the full string to match.